src/main/kotlin/no/nav/syfo/identhendelse/database/IdenthendelseQuery.kt

```diff
@@ -0,0 +1,113 @@
+
+const val queryUpdateKandidatStoppunkt =
+    """
+        UPDATE DIALOGMOTEKANDIDAT_STOPPUNKT
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateKandidatStoppunkt(nyPersonident: PersonIdentNumber, stoppunktWithOldIdent: List<PDialogmotekandidatStoppunkt>): Int {
+    var updatedRows = 0
+    this.connection.use { connection ->
+        connection.prepareStatement(queryUpdateKandidatStoppunkt).use {
+            stoppunktWithOldIdent.forEach { stoppunkt ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, stoppunkt.personIdent.value)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryUpdateKandidatEndring =
+    """
+        UPDATE DIALOGMOTEKANDIDAT_ENDRING
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateKandidatEndring(nyPersonident: PersonIdentNumber, endringWithOldIdent: List<PDialogmotekandidatEndring>): Int {
+    var updatedRows = 0
+    this.connection.use { connection ->
+        connection.prepareStatement(queryUpdateKandidatEndring).use {
+            endringWithOldIdent.forEach { endring ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, endring.personIdent.value)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryUpdateUnntak =
+    """
+        UPDATE UNNTAK
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateUnntak(nyPersonident: PersonIdentNumber, unntakWithOldIdent: List<PUnntak>): Int {
+    var updatedRows = 0
+    this.connection.use { connection ->
+        connection.prepareStatement(queryUpdateUnntak).use {
+            unntakWithOldIdent.forEach { unntak ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, unntak.personIdent)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryGetDialogmoteStatusCount =
+    """
+        SELECT COUNT(*)
+        FROM DIALOGMOTESTATUS
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.getDialogmoteStatusCount(
+    personIdent: PersonIdentNumber,
+): Int =
+    this.connection.use { connection ->
+        connection.prepareStatement(queryGetDialogmoteStatusCount).use {
+            it.setString(1, personIdent.value)
+            it.executeQuery().toList { getInt(1) }
+        }
+    }.firstOrNull() ?: 0
+
+const val queryUpdateDialogmoteStatus =
+    """
+        UPDATE DIALOGMOTESTATUS
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateDialogmoteStatus(nyPersonident: PersonIdentNumber, oldIdentList: List<PersonIdentNumber>): Int {
```

Comment:
Noen spesiell grunn til at man tar inn en liste med gamle identer her og ikke i de andre update-metodene over? Tenker det er unødvendig å sende inn liste med f.eks. PUnntak-objekter når vi bare er interessert i personidenten.

Reply:
Tja, ettersom at vi ikke hadde noen `PDialogmoteStatus`-klasse og ingen spørring for å hente ut dette, så gjør vi heller `SELECT COUNT(*) FROM DIALOGMOTESTATUS` i stedet for å gjøre om til database-domeneobjekt. Det fører da også til at vi i denne spørringen ikke har de spesifikke radene (kun antall rader - altså om identen finnes i databasen vs finnes ikke) og må derfor bruke `oldIdentList`. Det er ikke nødvendigvis sånn at alle identene i `oldIdentList` finnes i databasen vår, men vi vet at én av de gjør det.

Utenom det så kunne vi selvsagt også bare sendt liste med identer til de andre update-funksjonene også 🤔 Det er nok en forbedring sånn sett 👍🏼

Reply:
Skjønner. Ja, det blir i hvert fall mer konsistent API på Query-funksjonene om alle tar inn ident-liste og man slipper å lure på hvorfor det er ulikt 😄

Reply:
Enig i det 👍🏼